packages/core/src/lib/lit-core.ts

```diff
@@ -776,15 +805,45 @@ export class LitCore {
         this.latestBlockhash = blockHashBody.blockhash;
         this.lastBlockHashRetrieved = Date.now();
         log('Done syncing state new blockhash: ', this.latestBlockhash);
+
+        // If the blockhash retrieval failed, throw an error to trigger fallback in catch block
+        if (!this.latestBlockhash) {
+          throw new Error(
+            `Error getting latest blockhash. Received: "${this.latestBlockhash}"`
+          );
+        }
       })
-      .catch((err: BlockHashErrorResponse) => {
-        // Don't let error from this setInterval handler bubble up to runtime; it'd be an unhandledRejectionError
+      .catch(async (err: BlockHashErrorResponse | Error) => {
         logError(
-          'Error while attempting fetch new latestBlockhash:',
-          err.messages,
-          'reason: ',
-          err.reason
+          'Error while attempting to fetch new latestBlockhash:',
+          err instanceof Error ? err.message : err.messages,
+          'Reason: ',
+          err instanceof Error ? err : err.reason
         );
+
+        log(
+          'Attempting to fetch blockhash manually using ethers with fallback RPC URLs...'
+        );
+        const provider = await this._getProviderWithFallback();
+
+        if (!provider) {
+          logError(
+            'All fallback RPC URLs failed. Unable to retrieve blockhash.'
+          );
+          return;
```

Comment:
I think I found an edge case that can break. Follow me on this one:
- `fetch(this._blockHashUrl)` at line 802 fails -> we get into the `catch` clause
- the `catch` clause cannot get a provider or the one it gets cannot set the `latestBlockhash` (exiting at this line or failing at 837 `await provider.getBlock('latest')`)

If that happens (unstable providers/network might be a reason) then `_syncBlockhash` does not update `latestBlockhash`. The problem arises when considering that neither it nor its caller `getLatestBlockhash` discarded the old value of `latestBlockhash`.
So, if we already had a `latestBlockhash`, that might be outdated, then on `getLatestBlockhash` (line 450+) we might be not returning the real latest blockhash but an outdated one

It might not be such an issue as I think nodes still accept a wide range of blockhashes. Anyway, I think this would be easily fixed by setting `this.latestBlockhash = null` at the beginning of `_syncBlockhash`

Reply:
I think Instead of setting this.latestBlockhash to null immediately, we can check if the existing blockhash is outdated. If the blockhash is outdated, then proceed with the logic to fetch a new one. This prevents clearing a valid blockhash if it's still current and avoids using an outdated blockhash when it fails to sync. (we can re-use the code commented-out at 867)

so adding this in the beginning:
```
  private async _syncBlockhash() {
    const currentTime = Date.now();
    const blockHashValidityDuration = BLOCKHASH_SYNC_INTERVAL;

    if (
      this.latestBlockhash &&
      this.lastBlockHashRetrieved &&
      currentTime - this.lastBlockHashRetrieved < blockHashValidityDuration
    ) {
      log('Blockhash is still valid. No need to sync.');
      return;
    }
```

Reply:
I like it 🚀